pkg/telemetry/data.go

```diff
@@ -0,0 +1,78 @@
+package telemetry
+
+import (
+	"encoding/json"
+	"net/url"
+)
+
+// BaseData object definition containing the base data and it's mapping information
+type BaseData struct {
+	ActionName      string `json:"action_name"`
+	EventType       string `json:"event_type"`
+	SiteID          string `json:"idsite"`
+	URL             string `json:"url"`
+	StepName        string `json:"e_3"`
+	StageName       string `json:"e_10"`
+	PipelineURLHash string `json:"e_4"` // defaults to sha1 of env.JOB_URl
+	BuildURLHash    string `json:"e_5"` // defaults to sha1 of env.BUILD_URL
+}
+
+var baseData BaseData
+
+// BaseMetaData object definition containing the labels for the base data and it's mapping information
+type BaseMetaData struct {
+	StepNameLabel        string `json:"custom3"`
+	StageNameLabel       string `json:"custom10"`
+	PipelineURLHashLabel string `json:"custom4"`
+	BuildURLHashLabel    string `json:"custom5"`
+}
+
+// baseMetaData object containing the labels for the base data
+var baseMetaData BaseMetaData = BaseMetaData{
+	StepNameLabel:        "stepName",
+	StageNameLabel:       "stageName",
+	PipelineURLHashLabel: "pipelineUrlHash",
+	BuildURLHashLabel:    "buildUrlHash",
+}
+
+// CustomData object definition containing the data that can be set by a step and it's mapping information
+type CustomData struct {
+	// values custom_11 - custom_25 & e_11 - e_25 reserved for library reporting
+	Custom1Label string `json:"custom26,omitempty"`
+	Custom2Label string `json:"custom27,omitempty"`
+	Custom3Label string `json:"custom28,omitempty"`
+	Custom4Label string `json:"custom29,omitempty"`
+	Custom5Label string `json:"custom30,omitempty"`
+	Custom1      string `json:"e_26,omitempty"`
```

Comment:
There are some magic numbers (26, 27, ...). Maybe it would be a good idea to explain why they are like they are ...

Reply:
done